### PR TITLE
Added a Python script for replicating player saves for all other maps.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
## Description of changes

Adds a Python script that copies saves for the specified map, for all other maps.

My Python skills are pretty dogshit, mainly from incidental stuff at work, so any suggestions/changes would be appreciated.

## Why and what will this PR improve
For use by Scav since they're trying out a new map and manually copying saves is hell.

## Authorship
Myself.

## Changelog
Nothing player-facing.

### CODE REVIEW

**Comment by MistakeNot4892** [comment]
Needs to also update `preferences.json`.

**Comment by noelle-lavenza** [comment]
I don't really like this solution. Stella's solution, overriding the proc to get the map save key to be the same for both maps, works a lot better IMO. I guess you'd still need to migrate it to that path so there's nothing wrong with the script itself, but still.

**Comment by MistakeNot4892** [comment]
Making the key the same for all maps means you're saniziting the saves against different criteria when the map changes. Tradeship saves would lose some loadout choices if you opened them on ministation.

This was also needed immediately yesterday, so I banged out an external solution instead of one that needed testing in  DM.

File: tools/ReplicateSaves/replicate_saves.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+		"modpack_testing",
+		"random_ruins",
+		"~mapsystem",
+		"~unit_tests"
+	]
+	for dir in os.scandir(mapdir):
+		if os.path.isdir(dir):
+			targetmap = dir.path
+			targetmap = targetmap.replace(mapdir + os.sep, "")
+			if not targetmap in ignoremaps and targetmap != mapname:
```
Comment (out-of-phaze) [inline tools/ReplicateSaves/replicate_saves.py]:
```suggestion
			if targetmap not in ignoremaps and targetmap != mapname:
```

File: tools/ReplicateSaves/replicate_saves.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	# Collect slot names for each user to update their preferences.json
+	for (root, dirs, files) in os.walk(scrapedir):
+		has_prefs = False
+		has_saves = []
+		for file in files:
+			if file == "preferences.json":
+				has_prefs = True
+				continue
+			match = re.match(filename_regex, file, re.I)
+			if not match:
```
Comment (out-of-phaze) [inline tools/ReplicateSaves/replicate_saves.py]:
```suggestion
			if match is None:
```

**Comment by noelle-lavenza** [comment]
> Making the key the same for all maps means you're saniziting the saves against different criteria when the map changes. Tradeship saves would lose some loadout choices if you opened them on ministation.
> 
> This was also needed immediately yesterday, so I banged out an external solution instead of one that needed testing in DM.

Both tradeship and the new map would have the same key on Scavstation so that they'd share the same saves, and presumably they'd already be compatible anyway. I didn't mean make it the same for *all* maps.

**Review by out-of-phaze — DISMISSED** [review]
Could be cleaned up a bit but I don't see any actual blocking issues.

**Review by out-of-phaze — APPROVED** [review]
Assuming you've tested it. If so, looks good.

**Comment by MistakeNot4892** [comment]
Tested and working yep.

**Review by afterthought2 — APPROVED** [review]
I don't love the concept--saves are likely to break when you do this, or at any rate there is no guarantee that they won't. But if there is demand for it, OK I guess.

**Pull request merged**